src/idl_parser.cpp

```diff
@@ -2407,14 +2407,17 @@ CheckedError Parser::ParseEnum(const bool is_union, EnumDef **dest) {
   }
 
   if (dest) *dest = enum_def;
-  types_.Add(current_namespace_->GetFullyQualifiedName(enum_def->name),
-             new Type(BASE_TYPE_UNION, nullptr, enum_def));
+  if (types_.Add(current_namespace_->GetFullyQualifiedName(enum_def->name),
+                 new Type(BASE_TYPE_UNION, nullptr, enum_def)))
+    return Error("datatype already exists: " + enum_def->name);
   return NoError();
 }
 
 CheckedError Parser::StartStruct(const std::string &name, StructDef **dest) {
   auto &struct_def = *LookupCreateStruct(name, true, true);
-  if (!struct_def.predecl) return Error("datatype already exists: " + name);
+  auto qualified_name = current_namespace_->GetFullyQualifiedName(name);
+  if (!struct_def.predecl)
+    return Error("datatype already exists: " + qualified_name);
```

Comment:
The `qualified_name` is used only in this conditional expression.
Is it possible to replace `qualified_name` by direct call of `current_namespace_->GetFullyQualifiedName(name)`?